bin/clone

```diff
@@ -38,29 +32,55 @@ def main():
     parser.add_argument(
         "--css-validate", help="set whether css validation is required", type=str_to_bool, default=None
     )
+    parser.add_argument("--path", help="path to save the page to be cloned", required=False, default="/opt/")
+
     args = parser.parse_args()
+    default_path = os.path.join(args.path, 'snare')
+
     if args.log_path:
-        log_file = args.log_path + "clone.log"
+        log_file = os.path.join(args.log_path, "clone.log")
     else:
-        log_file = "/opt/snare/clone.log"
+        log_file = os.path.join(default_path, 'clone.log')
+
+    try:
+        if not os.path.exists(default_path):
+            os.makedirs(default_path)
+        with open(os.path.join(default_path, 'temp'), 'w') as tempfile:
+            tempfile.write('')
+        os.remove(os.path.join(default_path, 'temp'))
+
+        if not os.path.exists(os.path.dirname(log_file)):
+            os.makedirs(os.path.dirname(log_file))
+        with open(os.path.join(os.path.dirname(log_file), 'temp'), 'w') as tempfile:
+            tempfile.write('')
+        os.remove(os.path.join(os.path.dirname(log_file), 'temp'))
+    except IOError as e:
+        if os.getuid() != 0:
```

Comment:
this check is now redundant because you have already tested your permissions for the folders